lib/rouge/lexers/gdscript.rb

```diff
@@ -0,0 +1,201 @@
+# -*- coding: utf-8 -*- #
+# frozen_string_literal: true
+
+module Rouge
+  module Lexers
+    class GDScript < RegexLexer
+      title "GDScript"
+      desc "The Godot Engine programming language (https://godotengine.org/)"
+      tag 'gdscript'
+      aliases 'gd', 'gdscript'
+      filenames '*.gd'
+      mimetypes 'text/x-gdscript', 'application/x-gdscript'
+
+      affix_long  = '[rR]|[uUbB][rR]|[rR][uUbB]|[@]'
+      affix_short = '[uUbB]?'
+
+      keywords = %w(
+        and in not or as breakpoint class class_name extends is func setget
+        signal tool const enum export onready static var break continue
+        if elif else for pass return match while remote master puppet
+        remotesync mastersync puppetsync
+      ).join('|')
+
+      # Reserved for future implementation
+      keywords_reserved = %w(
+        do switch case
+      ).join('|')
+
+      buildins = %w(
+        Color8 ColorN abs acos asin assert atan atan2 bytes2var ceil char
+        clamp convert cos cosh db2linear decimals dectime deg2rad dict2inst
+        ease exp floor fmod fposmod funcref hash inst2dict instance_from_id
+        is_inf is_nan lerp linear2db load log max min nearest_po2 pow
+        preload print print_stack printerr printraw prints printt rad2deg
+        rand_range rand_seed randf randi randomize range round seed sign
+        sin sinh sqrt stepify str str2var tan tan tanh type_exist typeof
+        var2bytes var2str weakref yield
+      ).join('|')
+
+      builtins_type = %w(
+        bool int float String Vector2 Rect2 Transform2D Vector3 AABB
+        Plane Quat Basis Transform Color RID Object NodePath Dictionary
+        Array PoolByteArray PoolIntArray PoolRealArray PoolStringArray
+        PoolVector2Array PoolVector3Array PoolColorArray null
+      ).join('|')
+
+      state :root do
+        rule %r/\n/, Text
+        rule %r/^(\s*)([rRuUbB]{,2})("""(?:.|\n)*?""")/ do
+          groups Text, Str::Affix, Str::Doc
+        end
+        rule %r/^(\s*)([rRuUbB]{,2})('''(?:.|\n)*?''')/ do
+          groups Text, Str::Affix, Str::Doc
+        end
+        rule %r/[^\S\n]+/, Text
+        rule %r/#.*$/, Comment::Single
+        rule %r/[\[\]{}:(),;]/, Punctuation
+        rule %r/\\\n/, Text
```

Comment:
Am pretty sure this isn't necessary. The lexer already matches new lines (see line 61) and backslashes (see line 73). I realise this is from the Python lexer but I think it's a mistake. I can't see why it would be necessary to have all three rules.

Reply:
I think I understand the point of this rule. It's sometimes the case that scripts will split code across multiple lines by ending the line in a `\` to indicate that the parser should continue onto the next line. Do you know if GDScript works that way?

Reply:
Yeah, it's possible to use `\` like that in GDScript.

```gdscript
var t = "Hello, " \
        + "very long " \
        + "text"
```

Reply:
Yeah, my bad. Thanks for adding this back!